### PR TITLE
Features/#23 workflow description

### LINKED ISSUE #23: Describe the workflow of data processing
Use the section [workflow]() for describing how the eGo^n data processing works. It should include at least (please add more)

- [x] A  transparent and reproducible open data based data processing pipeline for generating scenario data...
- [x] Referring to ego^n project context (use ego-n.org)
- [x] Processes spatial data, time series and other data
- [x] In principle language-agnostic but mostly Python and SQL is used
- [x] Local containerized PostgreSQL database for SQL processing
- [x] Airflow organizes the order of execution of processing steps through so-called Operators
- [x] Updating [the workflow sketch](https://gist.github.com/gplssm/31aa4da15d58c3471f2398d29613363f) and including it would be help but maybe too much work!?
- [x] Airflow description
- [ ] Docker description including connection infos

### PR DESCRIPTION
**Pull request by IlkaCu**
Fix #23

### CODE REVIEW

File: docs/workflow.rst [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+
+The egon-data workflow is composed of four different sections: database setup, data import, data processing and data export to the OpenEnergy Platform. Each section consists of different tasks, which are managed by Apache Airflow and correspond with the local database. 
+Only final datasets which function as an input for the optimization tools or selected interim results are uploaded to the `Open Energy Platform <https://openenergy-platform.org/>`_. 
+The data processing in egon-data needs to be performed locally as calculations on the Open Energy Platform are prohibited. 
+
+.. _DP_workflow_sketch:
+.. figure:: images/DP_Workflow_15012021.svg
+ 
+
```
Comment (gplssm) [inline docs/workflow.rst:28]: Link to https://egon-data.readthedocs.io/en/latest/getting_started.html#run-the-workflow

File: docs/workflow.rst [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+Versioning
+----------
+
+Source code and data are versioned independendly from each other. Every data table uploaded to the Open Energy Platform contains a column 'version' which is used to identify different versions of the same data set. The version number is maintained for every table separately. This is a major difference to the versioning concept applied in the former data processing where all (interim) results were versioned under the same version number.
```
Comment (gplssm) [inline docs/workflow.rst:32]: Actually, this is not reality yet. Maybe we should mention that this isn't implement at the moment

**Review by gplssm — CHANGES_REQUESTED** [review]
I think this is a good start for the workflow description section for now. Please see my comments along the changed files.

And please, add a bullet point in the CHANGELOG to describe what you've added.

File: docs/workflow.rst [hunk 3]
```diff
@@ -1,3 +1,40 @@
...
+-----------------
+
+egon-data provides a transparent and reproducible open data based data processing pipeline for generating data models suitable for energy system modeling. The data is customized for the requirements of the research project eGo_n. The research project aims to develop tools for an open and cross-sectoral planning of transmission and distribution grids. For further information please visit the `eGo_n project website <https://ego-n.org/>`_.
+egon-data is a further development of the `Data processing <https://github.com/openego/data_processing>`_ developed in the former research project `open_eGo <https://openegoproject.wordpress.com/>`_. It aims for an extensions of the data models as well as for a better replicability and manageability of the data preparation and processing. 
+
+Data
+----
+
+egon-data retrieves and processes data from several different external input sources which are all freely available and published under an open data license. The process handles data with different data types, such as spatial data with a high geographical resolution or load/generation time series with an hourly resolution.  
+
```
[the remark this answers is not in the document]
Comment (IlkaCu) [reply docs/workflow.rst:16]: I added information on the further use of the egon-data results in this commit ca8d90d0f5447db479b37a4ceb04695c445fd09e

File: docs/workflow.rst (hunk 1 above)
Comment (IlkaCu) [reply docs/workflow.rst:30]: Done in 0ca836d3e6343a5f376aae91edfb940a8008f632

File: docs/workflow.rst (hunk 2 above)
Comment (IlkaCu) [reply docs/workflow.rst:36]: I included a short notification that the versioning is not yet implemented: e060d0afcc436005ba642a9ece2641bd009d78d6

**Comment by IlkaCu** [comment]
@gplssm: Thank you for your review. I included the changes you requested. Is it okay like this?

File: docs/workflow.rst (hunk 2 above)
Comment (gplssm) [reply docs/workflow.rst:37]: :+1: I put it into a warning box in 43b7cac68a3445f6eee4f5a455579c73fd6ab8bc

File: docs/workflow.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (gplssm) [reply docs/workflow.rst:16]: In addition to eTraGo and eDisGo, I added ding0 as tool that use eGon-data results 3eda4b8ab0def3e8d3b3b3a9a4b248b5cb5e3ff9

**Review by gplssm — APPROVED** [review]

**Pull request merged**